mantidproject/mantid
# Add Transmission data to XML and H5 Files

**Description of work.**
Added unfitted transmission sample and can data to XML (CanSAS algorithm) and H5 (NXCanSAS) output files if they are provided. Add unfitted sample/can workspaces to the unsubtracted sample/can files if "Save Can" option selected.

This brings v2 GUI inline with v1

**Report to:** steve king

**To test:**
1. Interfaces -> SANS -> SANS v2 GUI
2. In the provided data, load the SANS2D user and batch files
3. At the bottom of the screen, click the "file" radio button, check "Save Can" and check both "NXCanSAS" and "CanSAS" save options
4. In manage directories, select an output location
5. Click process
6. In your output folder, view the h5 and xml files created
7. The xml files should have a tag for `<SAStransmission_spectrum name="sample">`
8. In the h5 file, in sasentry01 there should be a group titled "sastransmission_spectrum_sample"

[PRdata.zip](https://github.com/mantidproject/mantid/files/2802616/PRdata.zip)

Fixes #24424 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
